lib/common-types.ts

```diff
@@ -1,91 +1,126 @@
 import { AxiosInstance } from 'contentful-sdk-core'
 import { Document as RichTextDocument } from '@contentful/rich-text-types'
+
 export type HttpClientInstance = AxiosInstance
+
 export interface AssetDetails {
   size: number
   image?: {
     width: number
     height: number
   }
 }
+
 export interface AssetFile {
   url: string
   details: AssetDetails
   fileName: string
   contentType: string
 }
+
 export interface AssetFields {
   title: string
   description: string
   file: AssetFile
 }
+
 export interface Asset {
-  sys: Sys
+  sys: AssetSys
   fields: AssetFields
 }
+
 export interface ContentfulCollection<T> {
   total: number
   skip: number
   limit: number
   items: Array<T>
 }
+
 export type AssetCollection = ContentfulCollection<Asset>
+
 export interface Entry<T> {
   sys: Sys
   fields: T
 }
+
 export interface EntryCollection<T> extends ContentfulCollection<Entry<T>> {
   errors?: Array<any>
   includes?: any
 }
+
 export interface ContentType {
-  sys: Sys
+  sys: ContentTypeSys
   name: string
   description: string
   displayField: string
   fields: Array<Field>
 }
+
 export declare type ContentTypeCollection = ContentfulCollection<ContentType>
+
 export interface Locale {
   code: string
   name: string
   default: boolean
   fallbackCode: string | null
-  sys: {
-    id: string
-    type: 'Locale'
-    version: number
-  }
+  sys: LocaleSys
 }
+
 export declare type LocaleCollection = ContentfulCollection<Locale>
+
 export interface SyncCollection {
   entries: Array<Entry<any>>
   assets: Array<Asset>
   deletedEntries: Array<Entry<any>>
   deletedAssets: Array<Asset>
   nextSyncToken: string
 }
-export interface Sys {
+
+interface BaseSys {
   type: string
   id: string
+}
+
+type EntitySys = {
   createdAt: string
   updatedAt: string
-  locale: string
   revision?: number
-  space?: {
-    sys: SpaceLink
-  }
-  contentType: {
-    sys: ContentTypeLink
-  }
+  space?: { sys: SpaceLink }
+  environment: { sys: EnvironmentLink }
+  locale: string
+} & BaseSys
+
+export interface Sys extends EntitySys {
+  contentType: { sys: ContentTypeLink }
 }
-export interface Link<T extends string> {
+
+export type AssetSys = EntitySys
+
+export type SpaceSys = { type: 'Space' } & BaseSys
+
+export type ContentTypeSys = {
```

Comment:
The pattern in the preceding types was to use `interface Foo extends Bar`.
Maybe we should do that here in the new sys types  as well for consistency.